### PR TITLE
[CSM] Fix syscall based drift events

### PR DESCRIPTION
**Pull request by Gui774ume**
### What does this PR do?

This PR improves how we generate drift events for syscalls past the initial activity dump learning period. Leveraging the same aggregation / conditional send logique, we're now leveraging the same event type for both the syscalls collection events during the activity dump learning period, and the syscall based drift events.

### Motivation

This change is necessary to make syscall drift events work while ensuring we won't spam events because of poorly profiled workloads.

### Additional Notes

Although the list of syscalls is collected and recorded in dumps per process, their evaluation is based on a flattened list that contains all the syscalls made across all processes.

### Describe how to test/QA your changes

- Load a minimalistic profile, turn on drift events for `syscalls` and make sure a syscall anomaly detection event is sent:
  - after a new process exits
  - after a new process execve into something else
  - after 1 minute of making new (at least one new) or existing syscalls in the profile

### CODE REVIEW

**Review by cswatt — APPROVED** [review]
approving for documentation

File: pkg/security/secl/model/category.go [hunk 1]
```diff
@@ -38,7 +38,7 @@ func GetAllCategories() []EventCategory {
 // GetEventTypeCategory returns the category for the given event type
 func GetEventTypeCategory(eventType eval.EventType) EventCategory {
 	switch eventType {
-	case "exec", "signal", "exit", "fork", "anomaly_detection_syscall":
+	case "exec", "signal", "exit", "fork", "anomaly_detection_syscall", "syscalls":
```
Comment (paulcacheux) [inline pkg/security/secl/model/category.go:41]: Do we still need `"anomaly_detection_syscall"` ?

**Review by paulcacheux — APPROVED** [review]

File: pkg/security/secl/model/category.go (hunk 1 above)
Comment (Gui774ume) [reply pkg/security/secl/model/category.go:41]: Nop, good catch thanks !

**Pull request merged**